### ISSUE TITLE
Querier panic with `{job="fluxy/prose",__name__!=""}`

### ISSUE THREAD

**Issue by aaron7** [weaveworks/cortex#582]
When executing `{job="fluxy/prose",__name__!=""}` across a range, we see a `422 Unprocessable Entity` response.

`{"status":"error","errorType":"execution","error":"unexpected error"}`

We also see the querier panic.
```
level=error msg="parser panic: runtime error: invalid memory address or nil pointer dereference
panic(0x18ceb00, 0x281c420)
/usr/local/go/src/runtime/panic.go:489 +0x2cf
```

This does not occur with an instant query.

**Comment by leth** [comment]
This seems to be fixed in https://github.com/weaveworks/cortex/pull/583

**Comment by juliusv** [comment]
Yep, the `LazySeriesIterator` in the Prom1 Cortex is broken - I fixed this in the Prom2 port #583, which is hopefully soon ready for rollout.

**Comment by bboreham** [comment]
Closing as fixed in #583
